crates/astria-sequencer-types/src/sequencer_block_data.rs

```diff
@@ -97,12 +122,23 @@ impl SequencerBlockData {
 
     #[allow(clippy::type_complexity)]
     #[must_use]
-    pub fn into_values(self) -> (Hash, Header, Option<Commit>, HashMap<Namespace, RollupData>) {
+    pub fn into_values(
```

Comment:
This is a code smell. We should differentiate between a "raw" `SequencerBlockData` as a (de)serialization format (where it's ok to make all fields pub), and "verified" `SequencerBlockData` that upholds invariants (where you want to prevent anyone pulling the rug from under you). Permitting `clippy::type_complexity` proves that point.

My suggestion: keep the `SequencerBlockData` as is, but introduce a `RawSequencerBlockData` that has exactly the same fields, but comes with a `RawSequencerBlockData::verify(self) -> eyre::Result<SequencerBlockData>`. This way you can keep using `SequencerBlockData` everywhere as is.

Next, scrap `SequencerBlockData::into_values` and introduce `SequencerBlockData::into_raw`. This way users can do the following instead of destructuring tuples:

```rust
fn do_something(data: SequencerBlockData) {
    let RawSequencerBlockData {
        block_hash,
        ...
    } = data.into_raw();
    further_processing(block_has);
}
```

We then have an easier path forward to move this into `astria-proto`.

Reply:
yeah I didn't really like this either but wasn't sure if I should just make another type with all pub fields. will update!